File: EXAMEN/01.SERVIDOR/WS_JAVA_REST_Comercializadora/src/main/java/ec/edu/monster/util/DataInitializer.java
package ec.edu.monster.util;

import ec.edu.monster.JPAUtil;
import ec.edu.monster.model.Usuario;
import jakarta.persistence.EntityManager;
import jakarta.persistence.TypedQuery;
import jakarta.servlet.ServletContextEvent;
import jakarta.servlet.ServletContextListener;
import jakarta.servlet.annotation.WebListener;

@WebListener
public class DataInitializer implements ServletContextListener {

    @Override
    public void contextInitialized(ServletContextEvent sce) {
        EntityManager em = JPAUtil.getEntityManager();
        try {
            em.getTransaction().begin();

            TypedQuery<Long> q = em.createQuery(
                    "SELECT COUNT(u) FROM Usuario u WHERE u.username = :user",
                    Long.class
            );
            q.setParameter("user", "MONSTER");
            Long count = q.getSingleResult();

            if (count == 0L) {
                Usuario admin = new Usuario();
                admin.setUsername("MONSTER");

                // 🔐 Guardar hash de la contraseña MONSTER9
                admin.setPassword(PasswordUtil.hashPassword("MONSTER9"));

                admin.setRol("SUPERADMIN");
                admin.setActivo(true);
                em.persist(admin);
            }

            em.getTransaction().commit();
        } catch (Exception e) {
            if (em.getTransaction().isActive()) em.getTransaction().rollback();
            e.printStackTrace();
        } finally {
            em.close();
        }
    }

    @Override
    public void contextDestroyed(ServletContextEvent sce) { }
}

File: PROYECTO/01.SERVIDOR/WS_DOTNET_SOAP_Comercializadora/ec.edu.monster.utils/DataInitializer.cs
using System.Linq;
using WS_DOTNET_SOAP_Comercializadora.ec.edu.monster.model;

namespace WS_DOTNET_SOAP_Comercializadora.ec.edu.monster.utils
{
    public static class DataInitializer
    {
        public static void EnsureAdminUser()
        {
            const string adminUser = "MONSTER";
            const string adminPasswordPlain = "MONSTER9";

            using (var ctx = new ComercializadoraContext())
            {
                bool exists = ctx.Usuarios.Any(u => u.Username == adminUser);

                if (!exists)
                {
                    var admin = new Usuario
                    {
                        Username = adminUser,
                        Password = PasswordUtil.HashPassword(adminPasswordPlain),
                        Rol = "SUPERADMIN",
                        Activo = true
                    };

                    ctx.Usuarios.Add(admin);
                    ctx.SaveChanges();
                }
            }
        }
    }
}
